### ISSUE TITLE
help with booting and installation on NUC8i3BEH

### ISSUE THREAD

**Issue by zapphyre** [zearp/Nucintosh#114]
Hi Zearp!
I would _kindly_ like to ask you for a help with debugging boot process of the Osx installation with your EFI config.
I have downloaded OSX Ventura installer from official Apple page and installed in into the applications on my (hackintoshed) old iMac now running OSX Monterey. then I formatted usb as for GUID partition type:
```
/dev/disk2 (external, physical):
   #:                       TYPE NAME                    SIZE       IDENTIFIER
   0:      GUID_partition_scheme                        *61.5 GB    disk2
   1:                        EFI ⁨EFI⁩                     209.7 MB   disk2s1
   2:                  Apple_HFS ⁨Install macOS Ventura⁩   61.2 GB    disk2s2
``` 
created installation by: `sudo /Applications/Install\ macOS\ Ventura.app/Contents/Resources/createinstallmedia --volume /Volumes/usb /Applications/Install\ macOS\ Ventura.app --nointeraction` after it completed it looks like this:
``` 
% ls /Volumes/Install\ macOS\ Ventura 
098-26676-071.dmg.j132ap.im4m		098-26676-071.dmg.j230kap.im4m		098-26788-071.dmg.j214kap.im4m
098-26676-071.dmg.j137ap.im4m		098-26676-071.dmg.j680ap.im4m		098-26788-071.dmg.j215ap.im4m
098-26676-071.dmg.j140aap.im4m		098-26676-071.dmg.j780ap.im4m		098-26788-071.dmg.j223ap.im4m
098-26676-071.dmg.j140kap.im4m		098-26676-071.dmg.x86legacyap.im4m	098-26788-071.dmg.j230ap.im4m
098-26676-071.dmg.j152fap.im4m		098-26788-071.dmg.j132ap.im4m		098-26788-071.dmg.j230kap.im4m
098-26676-071.dmg.j160ap.im4m		098-26788-071.dmg.j137ap.im4m		098-26788-071.dmg.j680ap.im4m
098-26676-071.dmg.j174ap.im4m		098-26788-071.dmg.j140aap.im4m		098-26788-071.dmg.j780ap.im4m
098-26676-071.dmg.j185ap.im4m		098-26788-071.dmg.j140kap.im4m		098-26788-071.dmg.x86legacyap.im4m
098-26676-071.dmg.j185fap.im4m		098-26788-071.dmg.j152fap.im4m		BaseSystem
098-26676-071.dmg.j213ap.im4m		098-26788-071.dmg.j160ap.im4m		Firmware
098-26676-071.dmg.j214ap.im4m		098-26788-071.dmg.j174ap.im4m		Install macOS Ventura.app
098-26676-071.dmg.j214kap.im4m		098-26788-071.dmg.j185ap.im4m		Library
098-26676-071.dmg.j215ap.im4m		098-26788-071.dmg.j185fap.im4m		System
098-26676-071.dmg.j223ap.im4m		098-26788-071.dmg.j213ap.im4m		restore
098-26676-071.dmg.j230ap.im4m		098-26788-071.dmg.j214ap.im4m		usr
``` 
as you state, I have replaced `AirportItlwm.kext` for Ventura-related one from 'stuff' folder. I have used `EFI Agent` to mount usb's EFI partition and command+c/v the modified EFI folder.
also, I have installed OpenCore Configurator and validated EFI folder for CPU family Coffee Lake, OC 0.8.8 and corrected some records that it suggested to change. I am still getting:
```
Kernel->Add[1] discovers CPUFriendDataProvider.kext, but its Parent (CPUFriend.kext) is either placed after it or is missing!
CheckKernel returns 1 error!

Completed validating /Volumes/EFI/EFI/OC/config.plist in 1 ms. Found 1 issue requiring attention.
```
but CPUFriend.kext is definitely present in the OC/Kexts folder. I have try to disable `NVMeFix.kext` as I saw you recommending this in some older issues here. 
Basically I have try everything that I can analytically think of and I have tried many EFI repositories/configurations I found here, on the github with various OSX versions. I have also upgraded my bios to the newest one provided by official Intel web. I have found colliding information about enabling Legacy Boot but obviously I have tried everything with/without is, but I did not move further away from the message `A bootable device has not been detected`.
I would greately appreciate you guidance on how to debug this and move a bit forward as I am quite new to this.

Thank you for all your efforts for the hacking community!

**Comment by zearp** [comment]
> I have installed OpenCore Configurator 

Those kind of tools can break the config and also makes it impossible for me to help properly as I don't know what changes been made, what kexts been changed, etc etc - I can't help when the default EFI has been changed or modified as it introduces too many variables. But I didn't really read any problems in your post so it doesn't matter in this case but still, please try to refrain from using those kind of tools. The config file isn't that scary once you edited a few times with ProperTree and once its setup you rarely have to edit it again. Also no changes are required for my EFI, only serials etc have to replaced in the config, thats all.

The warning you see if just a load order thing, the config you validated loads the data provider kext (the config) before cpu friend itself. In the end it shouldn't matter but to get rid of the message just make sure the data driver kext is loaded after cpu friend itself.

Legacy boot is not supported by macOS and shouldn't be used unless you need to use a very old OS. It changes a lot which can amongst many things, can break acpi patches and such. It is best to leave legacy boot and legacy usb disabled.

I'm not too sure what your issue is  to be honest but the EFI on the repo has its configuration validated as well and I wouldn't put up a release version that didn't pass config validation. It is a good check.

```
Validating "config.plist" with 0.9.1 acidanthera/ocvalidate
NOTE: This version of ocvalidate is only compatible with OpenCore version 0.9.1!
Completed validating config.plist in 0 ms. No issues found.
```

No problems found. I don't really understand why some people (you're not the first) use gui utils on the EFI to "fix" and "update" things instead of using the release version as is. I test the releases by doing a full install and upgrade with it before posting. Please use those then installation is really as simple as downloading the EFI and copying it to the EFI partition of the installer.

If you keep getting a no bootable device message with a usb stick that you made with createinstallmedia and have copied the EFI folder to the EFI partition of then check the bios of that machine because the stick would be bootable and booting EFI is very simple. The Apple script formats and creates the partitions for you, all you have to do is copy the EFI folder to the EFI partition.

You can easily test if UEFI boot is working properly by formatting any usb stick as fat32 then just copy the EFI folder to it and it will boot and show the OpenCore menu/picker. If not also check your bios boot order, maybe it tries to boot from an empty internal disk before a usb stick? There are too many options and I can't really guide people thought the install or do more troubleshooting.

The guide is made with clean slating to prevent issues as much as possible and always give the same results when the steps are followed which is why for example includes a bios reset and then only lists what needs to be changed. I follow my own guide when testing new releases to make sure it still works and the last time I did this everything was good. That was for the current macOS version Ventura. I also tested the 13.3 beta and it both installs and boots the beta without issues. I've not run into booting issues in a very long time.

Good luck!

**Comment by zearp** [comment]
It sounds like UEFI boot is broken on your NUC. The only thing I can suggest is to reset their bios to factory defaults and try again. Setting up macOS in legacy mode is an option but not a solution to your problem of having a non working UEFI boot.

I would stay away from macOS distro's such as Oralia , there is no one EFI that works good on all computers. Each machine needs specific patches and settings. Sure it may sort of work but it's not proper and will never be as fast and stable as a customised EFI made for specific hardware. Oralia also knows this which is why they offer fixes/custom EFI for a fee.

You need to figure out why your NUC isn't booting properly from UEFI and once you fix that you can install using my EFI.

Good luck.

**Comment by zapphyre** [comment]
exactly. I would very like to figure that out or fix it rather call it. tho I'm tech savy I don't know much about UEFI and it's shell, but..
I got an idea that as long I can boot Oralia's usb, I might switch an EFI for yours and it _worked_! Boot menu showed up and I have installed Monterey. I even upgraded to Ventura and all works for second day now! Even wifi works, can't connect to BT device tho it finds it, but hey...finally! it's snappy great for photo editing which was my goal! Very pleased...
thank you for all your work and if someone reads this that might has some insights how to debug this UEFI issue (I find many having the same problem) I'll be glad to hear from you.

Thanks again!
